src/main/java/com/bokudos/bokudosserver/threads/GameThread.java

```diff
@@ -2,58 +2,95 @@
 
 import com.bokudos.bokudosserver.constants.ServerConfigurationConstants;
 import com.bokudos.bokudosserver.dtos.GameDTO;
-import com.bokudos.bokudosserver.dtos.PlayerPacketDTO;
-import com.bokudos.bokudosserver.dtos.ServerPacketDTO;
-import com.bokudos.bokudosserver.external.stagebuilder.v1.data.Tiles;
+import com.bokudos.bokudosserver.enums.AssetType;
+import com.bokudos.bokudosserver.external.stagebuilder.Tiles;
+import com.bokudos.bokudosserver.packets.in.PlayerUpdatePacket;
+import com.bokudos.bokudosserver.packets.out.Animation;
+import com.bokudos.bokudosserver.packets.out.EnemyAsset;
+import com.bokudos.bokudosserver.packets.out.PlayerAsset;
+import com.bokudos.bokudosserver.packets.out.ServerUpdatePacket;
+import com.bokudos.bokudosserver.packets.out.enums.Direction;
+import com.bokudos.bokudosserver.packets.out.enums.Movement;
+import com.bokudos.bokudosserver.physics.EnemyPhysics;
+import com.bokudos.bokudosserver.physics.PhysicsSettings;
+import com.bokudos.bokudosserver.physics.PlayerPhysics;
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.web.socket.TextMessage;
 import org.springframework.web.socket.WebSocketSession;
 
 import java.io.IOException;
-import java.util.ArrayList;
 import java.util.HashMap;
```

Comment:
What's the difference between the hashmap and map?

Reply:
Map is an interface, typically a lot of times in Java people declare maps like this for example:
`Map<String, Object> thisMap = new HashMap<>();`
Where Hashmap is the specific implenation of a Map, but Map is just the interface. The nice thing about declaring Map this way and then referencing them as Map, you can easily change the type of the map only where it is initiallized without changing it elsewhere. The HashMap is propbably the most basic type of Map in Java, there are also Hashtables, ConcurrentHashMaps, SortedMaps, and others that all have their different purposes. Hope that answers your question.